packages/react-native-platform-override/src/BatchingQueue.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ *
+ * @format
+ */
+
+/**
+ * Executes actions, attempting to group by a given key
+ */
+export default class BatchingQueue<TKey> {
+  private keyedQueues: Map<TKey, Array<() => Promise<void>>> = new Map();
+  private currentKey?: TKey;
+
+  enqueue<T>(key: TKey, action: () => Promise<T>): Promise<T> {
```

Comment:
>key [](start = 13, length = 3)

nit: perhaps a check for if key has a value, else the logic goes wrong..

Reply:
We have strictNullChecks enabled, so the key can't be undefined or null unless we explicitly allow it.